lib/chef/knife/tidy_backup_clean.rb

```diff
@@ -194,7 +194,7 @@ def fix_chef_sugar_metadata
         Dir[::File.join(tidy.backup_path, "organizations/*/cookbooks/chef-sugar*/metadata.rb")].each do |file|
           ui.stdout.puts "INFO: Searching for known chef-sugar problems when uploading."
           s = Chef::TidySubstitutions.new(nil, tidy)
-          version = s.cookbook_version_from_path(file)
+          version = tidy.cookbook_version_from_path(file)
```

Comment:
Need to do `File.dirname(file)` before passing it to `tidy.cookbook_version_from_path` which is expecting the path to end with the cookbook directory and not `metadata.rb` which it will have here.